pkgs/desktops/gnome-3/extensions/buildGnomeExtension.nix

```diff
@@ -0,0 +1,63 @@
+{ pkgs, lib, stdenv, fetchzip, python3 }:
+
+let
+
+  buildGnomeExtension = {
+    # Every gnome extension has a UUID. It's the name of the extension folder once unpacked
+    # and can always be found in the metadata.json of every extension.
+      uuid
+    , name
+    , pname
+    , description
+    # extensions.gnome.org extension URL
+    , link
+    , version
+    , sha256
+  }:
+
+  stdenv.mkDerivation {
+    inherit pname;
+    version = builtins.toString version;
+    src = fetchzip {
+      url = "https://extensions.gnome.org/extension-data/${
+          builtins.replaceStrings [ "@" ] [ "" ] uuid
+        }.v${builtins.toString version}.shell-extension.zip";
+      inherit sha256;
+      stripRoot = false;
+      # The download URL may change content over time. This is because the
+      # metadata.json is automatically generated, and parts of it can be changed
+      # without making a new release. We simply substitute the possibly changed fields
+      # with their content from when we last updated, and thus get a deterministic output
+      # hash.
+      extraPostFetch = let
+        # The replacement of \\u -> \u is to undo a workaround we do in update-extensions.py which does the opposite operation,
+        # because Nix can't handle unicode in JSON. The other substitutions are fallout from the previous.
```

Comment:
Just a note that it should be in the next release: https://github.com/NixOS/nix/pull/4368

Reply:
@andir told me (IIRC) that this was not enough, one cannot simply depend on the currently released version do to upgrade compatibility or something like that. Especially, all the build tooling (Hydra, Ofborgs) needs to be updated or otherwise things will break ^^. But if I get a "go" on that one, I'll be happy to change it.